ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Client.java

```diff
@@ -112,6 +112,48 @@
    */
   CompletableFuture<Void> terminatePushQuery(String queryId);
 
+  /**
+   * Sends a SQL request to the ksqlDB server. This method supports 'CREATE', 'CREATE ... AS
+   * SELECT', 'DROP', 'TERMINATE', and 'INSERT INTO ... AS SELECT' statements.
+   *
+   * <p>Each request should contain exactly one statement. Requests that contain multiple statements
+   * will be rejected by the client, in the form of failing the {@code CompletableFuture}, and the
+   * request will not be sent to the server.
+   *
+   * <p>The {@code CompletableFuture} is completed once a response is received from the server.
+   * Note that the actual execution of the submitted statement is asynchronous, so the statement
+   * may not have been executed by the time the {@code CompletableFuture} is completed.
+   *
+   * <p>If a non-200 response is received from the server, the {@code CompletableFuture} will be
+   * failed.
+   *
+   * @param sql the request to be executed
+   * @return a future that completes once the server response is received
+   */
+  CompletableFuture<Void> executeStatement(String sql);
```

Comment:
I considered having a return type to contain the command sequence number here, rather than returning void. The command seqeunce number could be useful in the future, since it could be used to ensure that subsequent requests (transient queries, admin operations, etc.) are not executed until the necessary streams/tables have been created, but this is advanced usage that is not currently supported by the client, so exposing the command sequence number right now is not useful. Is it worth introducing an empty return type, e.g., `ExecuteStatementResult` as a placeholder, in case we want to expose the command sequence number in the future? Not sure how confusion from having an empty return type now trades off against potentially updating the API in a backwards-incompatible way in the future.

Reply:
Update: once https://github.com/confluentinc/ksql/pull/5814 is merged, I'm going to update this return type to contain an `Optional<String>` containing the query ID, in the case of statements that start persistent queries. This is useful so client apps can easily terminate persistent queries.

Reply:
If we don't complete the future until the command has been executed then I don't think we need a command sequence number. It seems perhaps that the command sequence number on the REST API is a workaround as there's no way to get a callback on completion with REST. But with a proper future I think we can implement it in a nicer way :)

Reply:
> If we don't complete the future until the command has been executed then I don't think we need a command sequence number. It seems perhaps that the command sequence number on the REST API is a workaround as there's no way to get a callback on completion with REST. But with a proper future I think we can implement it in a nicer way :)

Yeah, this is a good point. As mentioned above, though, we unfortunately don't have a good way to tell when the statement has been executed, and it's also unclear exactly what that means since some servers may have executed the statement while others haven't. Maybe we could add a server endpoint to expose the command sequence number of the most recent command executed by the server? Though that still wouldn't address the issue of different servers having executed up through different statements.

Reply:
Yeah, sounds a bit complex and out of scope for now :)